lib/user.js

```diff
@@ -1,118 +1,65 @@
+const githubHelper = require('./github');
 const nodeify = require('./nodeify');
 const schema = require('screwdriver-data-schema');
-const Breaker = require('circuit-fuses');
-const github = new Github();
-
-/**
- * Github command to run
- * @method githubCommand
- * @param  {Object}   options            An object that tells what command & params to run
- * @param  {String}   options.token      Github token
- * @param  {String}   options.action     Github method. For example: get
- * @param  {Object}   options.params     Parameters to run with
- * @param  {Function} callback           Callback function from github API
- */
-function githubCommand(options, callback) {
-    github.authenticate({
-        type: 'oauth',
-        token: options.token
-    });
-    github.repos[options.action](options.params, callback);
-}
-
-const githubBreaker = new Breaker(githubCommand);
 
 class UserModel extends BaseModel {
 
     /**
      * Construct a UserModel object
      * @method constructor
-     * @param  {Object}    datastore         Object that will perform operations on the datastore
-     * @param  {String}    password          Login password
-     */
-    constructor(datastore, password) {
-        super('user', datastore);
-        this.password = password;
-    }
-
-    /**
-     * Create a user
-     * @method create
      * @param  {Object}   config                Config object to create the user with
+     * @param  {Object}   config.datastore      Object that will perform operations on the datastore
      * @param  {String}   config.username       The username
      * @param  {String}   config.token          The user's github token
-     * @param  {Function} [callback]            fn(err, data) where data is the new object created
-     * @return {Promise}                        If no callback is provided, a Promise is returned.
+     * @param  {String}   [config.password]     The user's password
      */
-    create(config, callback) {
-        const id = this.generateId(config);
-        const userConfig = {
-            table: this.table,
-            params: {
-                id,
-                data: config
-            }
-        };
-
-        return nodeify.withContext(this.datastore, 'save', [userConfig], callback);
+    constructor(config) {
+        super('user', config);
+        this.password = config.password;
     }
 
     /**
      * Seal token
      * @method sealToken
-     * @param  {String}   token      User's github token
-     * @param  {Function} [callback] fn(err, sealed) where sealed is the sealed token
-     * @return {Promise}             If no callback is provided, a Promise is returned.
+     * @param  {String}   token      Token to seal
+     * @return {Promise}
      */
-    sealToken(token, callback) {
-        return nodeify.withContext(iron, 'seal', [token, this.password, iron.defaults], callback);
+    sealToken(token) {
+        // TODO: update user model and datastore with new sealed token???
```

Comment:
Yes. Each user has a sealed token that's associated with him/her, and it should be stored in the `users` table. Previously we did this outside but with this refactoring I think it makes sense to put that logic here.